rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -470,15 +496,153 @@ private void mergeRequestedDependencies(List<Dependency> incomingRequestedDepend
+                    merged.add(dependency);
+                }
+            }
+
+            return merged;
+        }
+
+        private JsonNode mergePluginConfigurations(JsonNode configuration, JsonNode incomingConfiguration) {
+            ObjectNode ret = incomingConfiguration.deepCopy();
+            Iterator<Map.Entry<String, JsonNode>> fields = configuration.fields();
+            while (fields.hasNext()) {
+                Map.Entry<String, JsonNode> conf = fields.next();
+                JsonNode parentConf = ret.get(conf.getKey());
+                JsonNode parentCombine = parentConf.get("combine.children");
+                if (parentCombine != null && "append".equals(parentCombine.asText())) {
+                    JsonNode selfCombine = conf.getValue().get("combine.self");
+                    if (selfCombine != null && "override".equals(selfCombine.asText())) {
+                        ret.set(conf.getKey(), conf.getValue());
+                    } else {
+                        ret.set(conf.getKey(), combineLists(conf.getValue(), parentConf));
+                    }
+                } else {
+                    ret.set(conf.getKey(), conf.getValue());
+                }
+            }
+            return ret;
+        }
+
+        private JsonNode combineLists(JsonNode list, JsonNode incomingList) {
+            ObjectNode ret = incomingList.deepCopy();
+            ArrayList<String> keys = new ArrayList<>();
+            ret.fieldNames().forEachRemaining(keys::add);
+            keys.remove("combine.children");
+            // If no keys remaining, it's an empty list, we return the other one
+            if (keys.isEmpty()) {
+                return list.deepCopy();
+            }
+            // We can only have one key remaining in a list
+            String arrayElemField = keys.get(0);
+
+            // Copy elements of the list
+            JsonNode retNode = ret.get(arrayElemField);
+            JsonNode node = list.get(arrayElemField);
+            if (!(retNode instanceof ArrayNode)) {
+                ArrayNode arrayNode = JsonNodeFactory.instance.arrayNode();
+                arrayNode.add(retNode);
+                ret.set(arrayElemField, arrayNode);
+                retNode = arrayNode;
+            }
+            if (node instanceof ArrayNode) {
+                ((ArrayNode) retNode).addAll((ArrayNode) node);
+            } else if (node != null) {
+                ((ArrayNode) retNode).add(node);
+            }
+
+            // Check if combine.children is overridden
+            JsonNode listCombine = list.get("combine.children");
+            if (listCombine != null) {
+                ret.set("combine.children", listCombine);
+            }
+
+            return ret;
+        }
+
+        private List<Plugin.Execution> mergePluginExecutions(List<Plugin.Execution> executions, List<Plugin.Execution> incomingExecutions) {
+            return executions.isEmpty() ? incomingExecutions : executions; // TODO
```

Comment:
we are not going to use executions right now, but would be a good idea to implement this properly at some point.